zipline/gens/tradesimulation.py

```diff
@@ -22,6 +22,7 @@
     DATASOURCE_TYPE
 )
 from zipline.gens.utils import hash_args
+from qcommons.date_utils import utcnow
```

Comment:
qcommons is not available to Zipline.

This timing might be better done, for now at least, on the internal wrapper of handle_data.